snapcraft/internal/lifecycle.py

```diff
@@ -243,6 +244,18 @@ def _run_step(self, step, part, part_names):
         common.env.extend(self.config.project_env())
 
         part = _replace_in_part(part)
+
+        # Add the annotated list of build packages
+        part_build_packages = part._part_properties['build-packages']
+        part.build_packages = []
+        for pkg in self.build_packages:
```

Comment:
Right. Let me explain my confusion with an example:

```yaml
name: foo
# ...
build-packages: [bar]
parts:
  part1:
    plugin: nil
    build-packages: [baz]
```

At this point in the code, the build packages will have been installed, and self.build_packages will contain, say, `['bar=1.2', 'baz=3,4']`. Right? Then you loop over these, and compare each one to the build-packages for the part, which is just 'baz', and doesn't match anything. So then you split the package name and version apart, and place 'bar' and 'baz' into `part.build_packages`. Do I understand correctly?